src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -626,64 +691,56 @@ public Visitable visit(FieldDeclaration node, Void arg) {
+                node.getDeclarationAsString(false, false, false));
+    if (targetMethodsNames.contains(methodQualifiedSignature)) {
+      insideTargetMethod = true;
+    }
     addTypeVariableScope(node.getTypeParameters());
     Visitable result = super.visit(node, arg);
     typeVariables.removeFirst();
+    if (targetMethodsNames.contains(methodQualifiedSignature)) {
+      insideTargetMethod = false;
+    }
     return result;
   }
 
   @Override
   public Visitable visit(MethodDeclaration node, Void arg) {
-    // a MethodDeclaration instance will have parent node
-    Node parentNode = node.getParentNode().get();
-    Type nodeType = node.getType();
-
-    // This scope logic must happen here, because later in this method there is a check for
-    // whether the return type is a type variable, which must succeed if the type variable
-    // was declared for this scope.
-    addTypeVariableScope(node.getTypeParameters());
-
-    // since this is a return type of a method, it is a dot-separated identifier
-    @SuppressWarnings("signature")
-    @DotSeparatedIdentifiers String nodeTypeAsString = nodeType.asString();
-    @ClassGetSimpleName String nodeTypeSimpleForm = toSimpleName(nodeTypeAsString);
-    if (!this.isTypeVar(nodeTypeSimpleForm)) {
-      // Don't attempt to resolve a type variable, since we will inevitably fail.
-      try {
-        nodeType.resolve();
-      } catch (UnsolvedSymbolException | UnsupportedOperationException e) {
-        updateUnsolvedClassWithClassName(nodeTypeSimpleForm, false, false);
-      }
-    }
-
-    if (!insideAnObjectCreation(node)) {
-      SimpleName classNodeSimpleName = getSimpleNameOfClass(node);
-      className = classNodeSimpleName.asString();
-      methodAndReturnType.put(node.getNameAsString(), nodeTypeSimpleForm);
-    }
-    // node is a method declaration inside an anonymous class
-    else {
-      try {
-        // since this method declaration is inside an anonymous class, its parent will be an
-        // ObjectCreationExpr
-        ((ObjectCreationExpr) parentNode).resolve();
-      } catch (UnsolvedSymbolException | UnsupportedOperationException e) {
-        SimpleName classNodeSimpleName = ((ObjectCreationExpr) parentNode).getType().getName();
-        String nameOfClass = classNodeSimpleName.asString();
-        updateUnsolvedClassOrInterfaceWithMethod(
-            node, nameOfClass, toSimpleName(nodeTypeAsString), false);
-      }
+    String methodQualifiedSignature =
+        this.currentClassQualifiedName
+            + "#"
+            + TargetMethodFinderVisitor.removeMethodReturnType(
+                node.getDeclarationAsString(false, false, false));
+    String methodSimpleName = node.getName().asString();
+    if (targetMethodsNames.contains(methodQualifiedSignature)) {
+      // this approach will fail if target method a() is inside target method b(), but I don't see
+      // any reason why anyone would pass the arguments to Specimin that way.
```

Comment:
Can you add an input validation check for this in `SpeciminRunner`? It should be easy enough to check for overlap in the names and give the user a helpful error message if they ever do this.

Reply:
Professor,

I've just realized that I was overthinking. Right now Specimin does not support having a method inside an anonymous class as a target method, since Specimin determines the signature of a method base on the qualified name of the current non-anonymous class and the declaration of that method (as we can see in the code).